docs/design-docs/overview.md

```diff
@@ -0,0 +1,144 @@
+---
+id: overview
+---
+
+# Superbench Design
+
+## Goals
+
+Extend current tool to fit distribution test needs in cluster with 100 ~ 1000 nodes, make it modulable, easy to use and easy to scale up.
+Provide more simple and convenient way to:
+1.	Install and deploy the tool in a raw cluster, including bare metal, on-premises, and cloud environments.
+2.	Configure super benchmark configurations, by specifying config file or using command line arguments.
+3.	Launch super benchmark locally, on selected nodes, or on all nodes in the cluster. Use single device, selected devices, or all devices, including GPU and InfiniBand.
+4.	Be able to run with different distribution mode, e.g., mpi, torch distributed, ps/worker or client/server communication, etc.
+5.	Collect log during running and save results after running, merge results from all nodes to single one.
+6.	Provide a unified interface for all benchmarks, including how to run on different device vendor (NVIDIA/AMD), how to run on different mode (local/mpi), how to pass configurations and save results.
+
+## Design Principle
+
+1.	Make the tool a standard python package in pythonic way.
+2.	Only add necessary new features.
+3.	Keep frequently used benchmarks at the first stage.
+4.	Refactor code structure and internal interfaces, more features/benchmarks to be added in the future.
+
+## Architecture
+
+![SuperBench Workflow](../assets/executor_workflow.png)
+
+## Pipeline
+
+![Pipeline](../assets/executor-pipeline.png)
+
+1.	User prepares config file and host file. Both files could be omitted and use the default config or command line arguments to specify.
+2.	User runs sb cli on head node. Command line interface could accept a set of arguments and provide help information to user.
+3.	Sb cli parses the input config file, host file, and arguments, loads into one config object, and calls sb runner to start a run.
+4.	Sb runner parses the config object, and for all compute nodes in config,
+    1. Check the connection.
+    2. Check docker environment and sb Docker image.
+    3. Start an sb Docker container, mount necessary paths.
+    4. Prepare running context, including local package code, config file, SSH key pairs, SSH config for password less use. Tar all context and distribute to compute node.
+    5. Prepare output path.
+    6. Start SSH service and check inter-connections.
+5.	Sb runner loops all benchmarks and all modes in each benchmark. For each mode in each benchmark, sb runner calls sb executor inside Docker container on corresponding compute nodes to start an execution.
```

Comment:
`SuperBenchRunner` loops all benchmarks and all modes in each benchmark. For each mode in each benchmark, `SuperBenchRunner` calls `SuperBenchExecutor` inside Docker container on corresponding nodes to start an execution.